README.md

```diff
@@ -1,9 +1,205 @@
 [<img alt="libfabric master branch Travis CI status" src="https://travis-ci.org/ofiwg/libfabric.svg?branch=master"/>](https://travis-ci.org/ofiwg/libfabric)
 [<img alt="libfabric Coverity scan suild status" src="https://scan.coverity.com/projects/4274/badge.svg"/>](https://scan.coverity.com/projects/4274)
 
-libfabric
-=========
+# libfabric
 
-The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric communication services to applications.
+The Open Fabrics Interfaces (OFI) is a framework focused on exporting fabric
+communication services to applications.
 
-See [the OFI web site](http://ofiwg.github.io/libfabric/) for more details, including a description and overview of the project, and detailed documentation of the libfabric APIs.
+See [the OFI web site](http://libfabric.org) for more details, including a
+description and overview of the project, and detailed documentation of the
+libfabric APIs.
+
+## Building libfabric
+
+If you are building Libfabric from a developer Git clone, you must first run
+the `autogen.sh` script. This will invoke the GNU Autotools to bootstrap
+Libfabric's configuration and build mechanisms. If you are building Libfabric
+from an official distribution tarball from libfabric.org, there is no need to
+run `autogen.sh`; Libfabric distribution tarballs are already bootstrapped for
+you.
+
+Libfabric currently supports GNU/Linux, Free BSD, and OS X.
+
+### Dependencies
+
+- `libpthread`
+- `libdl`
+
+Provider dependencies will be discussed in the the "Providers" section.
+
+### Configure options
+
+Configure ships with many built in options (see `./configure --help`). Some
+useful options are:
+
+```
+--prefix=<directory>
+```
+
+By default `make install` will place the files into a system wide location.
+The prefix option specifies that the fabtests files should be installed into
+the directory named `<directory>`. The executables will be located at
+`<directory>/bin`.
+
+```
+--with-valgrind=<directory>
+```
+
+Directory where valgrind is installed.If valgrind is found, then valgrind
+annotations are enabled. This may incur a performance penalty.
+
+```
+--enable-debug
+```
+
+Enable debug code paths. This enables various extra checks and allows for using
+the highest verbosity logging output that is normally compiled out in
+production builds.
+
+```
+--enable-<provider>=[yes|no|auto|dl|PATH]
+--disable-<provider>
+```
+
+This enables or disables the provider named `<provider>`. Each provider is
+enabled by default. The `dl` option builds the provider as a loadable library
```

Comment:
"Each provider is enabled by default" -- not quite correct.

The default behavior is "auto" (which is not described here).  You might want to make a bullet list with a bullet for each option, and describe what each does.